syntax = "proto3";

package chat;

// Messages

message RegisterRequest {
  string name = 1;
}

message RegisterResponse {
  int32 id = 1;
}

message DirectoryRequest {
  int32 id = 1;
}

message DirectoryResponse {
  repeated User users = 1;
}

message User {
  int32 id = 1;
  string name = 2;
}

message RouteRequest {
  int32 sender_id = 1;
  int32 receiver_id = 2;
  string message = 3;
}

message RouteResponse {
  bool success = 1;
}

// Services

service ChatService {
  rpc RegisterUser(RegisterRequest) returns (RegisterResponse);
  rpc GetDirectory(DirectoryRequest) returns (DirectoryResponse);
  rpc RouteMessage(RouteRequest) returns (RouteResponse);
}
